### PR TITLE
[v9] Remove GHA workflows that test with `--use-libraries`

### PR DESCRIPTION
**Pull request by ncooke3**
#no-changelog

### CODE REVIEW

File: .github/workflows/messaging.yml [hunk 1]
```diff
@@ -148,16 +148,15 @@ jobs:
...
+          '--use-modular-headers'
         ]
     needs: pod-lib-lint
     steps:
     - uses: actions/checkout@v2
     - name: Setup Bundler
       run: scripts/setup_bundler.sh
     - name: PodLibLint Messaging Cron
-      run: scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseMessaging.podspec --platforms=${{ matrix.target }}
+      run: scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseMessaging.podspec --platforms=${{ matrix.target }} --allow-warnings
```
Comment (paulb777) [inline .github/workflows/messaging.yml:159]: Why is ` --allow-warnings` added? Are the warnings fixable?

**Review by paulb777 — COMMENTED** [review]
Thanks for working through this one!  Let's resolve the warning question and then good to go.

File: .github/workflows/messaging.yml (hunk 1 above)
Comment (ncooke3) [reply .github/workflows/messaging.yml:159]: Yep, so the messaging tests have an [intentional warning](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseMessaging/Tests/UnitTestsSwift/FIRMessagingAPITest.swift#L44-L52) that [fails a linting](https://github.com/firebase/firebase-ios-sdk/runs/5591176589?check_suite_focus=true).

Comment (paulb777) [reply .github/workflows/messaging.yml:159]: Hmmm, Maybe we should run two pod lib lints so we can still catch warnings in the library?

One with `--skip-tests` and one with `--allow-warnings`

**Review by paulb777 — APPROVED** [review]

File: .github/workflows/messaging.yml (hunk 1 above)
Comment (ncooke3) [reply .github/workflows/messaging.yml]: Discussed offline. Added `@unkown default` to silence warning. Resolved in latest work.

**Pull request merged**